tests/Feature/ApplicantControllerTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Applicant;
+use App\Models\Classification;
+use App\Models\Lookup\CitizenshipDeclaration;
+use App\Models\Lookup\VeteranStatus;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class ApplicantControllerTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * Base route for the API calls.
+     *
+     * @var string
+     */
+    protected $baseUrl;
+
+    /**
+     * Run parent setup and set global values.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->baseUrl = 'api/v1';
+        $this->faker = \Faker\Factory::create();
+    }
+
+    public function testGetProfile(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+
+        $response = $this->actingAs($applicant->user)
+            ->getJson("$this->baseUrl/applicant/$applicant->id/profile");
+        $response->assertOk();
+        $response->assertJsonFragment([
+            'citizenship_declaration_id' => $applicant->citizenship_declaration_id,
+            'veteran_status_id' => $applicant->veteran_status_id,
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ]);
+    }
+
+    public function testUpdateProfile(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $applicant->applicant_classifications()->delete();
+
+        // Ensure that submitting applicant classifications creates new elements.
+        $newApplicantClassification = [
+            'id' => 12345,
+            'applicant_id' => $applicant->id,
+            'classification_id' => Classification::inRandomOrder()->first()->id,
+            'level' => 5,
+            'order' => 1,
+        ];
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [
+                0 => $newApplicantClassification
+            ],
+        ];
+        $response = $this->actingAs($applicant->user)
+            ->putJson("$this->baseUrl/applicant/$applicant->id/profile", $updateData);
+        $response->assertOk();
+
+        $expectedData = [
+            'citizenship_declaration_id' => $updateData['citizenship_declaration_id'],
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ];
+        $response->assertJsonFragment($expectedData);
+        $this->assertDatabaseHas(
+            'applicant_classifications',
+            $applicant->applicant_classifications->first()->toArray()
```

Comment:
This should fail as is, I think, as I'd expect that the `applicant_classification` in the database has a different id from what we sent to api.